### PR TITLE
Redesign of FileWidget

### PR DESCRIPTION
**Pull request by creviera**
# Description

Resolves https://github.com/freedomofpress/securedrop-client/issues/525

file-widget

Since export and print are not yet supported in the client, I added code to temporarily hide these options so the client shows the file name all the way to the left.

When adding support for Export and Print there will need to be some FileWidget polishing.

### CODE REVIEW

**Comment by creviera** [comment]
Oh, I forgot to mention that in the screenshot above I have an example where there is a file of size `874 bytes` which takes us past the maximum file size column width. There could be file sizes that produce an even larger string. I think it looks better to shorten the horizontal line separator when the file size gets past a certain size, but would like @ninavizz's feedback!

File: securedrop_client/gui/widgets.py [hunk 1]
```diff
@@ -716,7 +716,8 @@ class EmptyConversationView(QWidget):
 
     CSS = '''
     #content {
-        font-family: Orbitron;
```
Comment (creviera) [inline securedrop_client/gui/widgets.py]: this was leftover from a manual test

File: securedrop_client/gui/widgets.py [hunk 2]
```diff
@@ -1474,53 +1477,16 @@ def _update_text(self, message_id: str, text: str) -> None:
             self.message.setText(text)
 
 
-class ConversationWidget(QWidget):
```
Comment (creviera) [inline securedrop_client/gui/widgets.py]: was able to achieve this by passing alignment when adding widgets to the layout (removes code [yay] and simplifies what MessageWidgets and ReplyWidgets are: they are SpeechBubbles, not ConversationWidgets that have SpeechBubbles)

**Comment by ninavizz** [comment]
Hey Allie, thanks for flagging! Let's use a library for filesizes that utilizes the following formats, to keep the strings short. If the below lowercase/uppercase designations aren't possible, I'm fine with it all being uppercase: 
**Bytes** =`123b` 
**KB** =`123kb` 
**Megabytes** =`123MB`

Also, I updated [the spec mockup in Zeplin](https://zpl.io/aw6K50n), to reflect the above. :)

**Comment by redshiftzero** [comment]
heads up for the interested observer, we discussed the formatting for file sizes today synchronously and decided to stick with B, kB, MB due to avoid any bits (indicated via lowercase b) versus bytes (indicated by uppercase B) confusion by the user (hence 0d91a8393045e1eb96aa194b782b9e2d4e42bab6)

File: securedrop_client/gui/widgets.py [hunk 3]
```diff
@@ -1588,44 +1619,109 @@ def __init__(
...
+        self.print_button.setObjectName('export_print')
+        self.print_button.setFont(file_buttons_font)
+        file_options_layout.addWidget(self.download_button)
+        file_options_layout.addWidget(self.export_button)
+        file_options_layout.addWidget(self.print_button)
+
+        # File name or default string
+        self.file_name = SecureQLabel(self.file.original_filename)
+        self.file_name.setObjectName('file-name')
+        self.no_file_name = QLabel('ENCRYPTED FILE ON SERVER')
```
Comment (redshiftzero) [inline securedrop_client/gui/widgets.py]: another nit but thought for the future: we may as well use `SecureQLabel` basically everywhere `QLabel` is used, the reason being that right now this string is hardcoded by the developer but when we localize these strings, a translator could attempt to inject HTML here

File: securedrop_client/gui/widgets.py [hunk 4]
```diff
@@ -1570,14 +1543,72 @@ def _on_reply_failure(self, message_id: str) -> None:
...
+        background-color: rgba(211, 216, 234, 0.45);
+        min-height: 2px;
+        max-height: 2px;
+        margin: 0px 8px 0px 8px;
+    }
+    '''
+
+    TOP_MARGIN = 10
+    FILE_FONT_SPACING = 2
+    FILE_OPTOINS_FONT_SPACING = 1.6
```
[the remark this answers is not in the document]
Comment (creviera) [reply securedrop_client/gui/widgets.py]: ty

**Comment by creviera** [comment]
note: this includes some fine-tuning work around speech bubbles, the conversation view, scrollarea, conversation header, reply box, conversation wrapper, and main view - there will be followup work for each individual section of the GUI, but since the UI is so interconnected with CSS styling, it's hard to avoid other areas of the client

double note: i know longer see the bug where the last item in the sourcelist would disappear, so resolving the issue below

resolves https://github.com/freedomofpress/securedrop-client/issues/331

**Pull request merged**